### DISCUSSION TITLE
Lock mechanism with servo

### DISCUSSION THREAD

**Discussion by markazik** [HomeSpan/HomeSpan#500] [Q&A]
Hi,

I'd like to setup a lock with esp32servo library. Simply setting the servo in two states 0 and 180 when lock and unlocked. In documentation I see two characteristics for leckmechanism - LockTargetState and LockCurrentState. How to integrate this with the servo as mentioned above? Any help would be appreciated as I can't find any examples for locks.

Thanks!

**Comment by HomeSpan** [comment]
See Example Tutorial 13 for details on how currentState and targetState Characteristics work.  Also note that HomeSpan has a built-in [Servo PWM class](https://github.com/HomeSpan/HomeSpan/blob/master/docs/PWM.md) designed to work nicely with the library.  The esp32servo library may interfere with HomeSpan.

**Comment by markazik** [comment]
Thanks for your help. Tutorial 13 gives some thoughts how to approach this but I still don't understand how to use Servo PWM in my scenario - lock mechanism and moving servo into two angles 0 and 180 degrees. Im little bit lost ;)
